packages/xstate-graph/src/graph.ts

```diff
@@ -234,7 +238,12 @@ export function traverseShortestPlans<TState, TEvent extends EventObject>(
         const { state: nextState, event: eventObject } = adjacency[
           serializedState
         ].transitions[event];
-        const nextSerializedState = serializeState(nextState, eventObject);
+        const prevState = stateMap.get(serializedState);
```

Comment:
Thankfully, the `stateMap` contains a mapping of serialized states to their actual `TState` values, so we can reference them.

In the future, this should be a built-in feature of the adjacency map object (i.e., let's make the object smarter and more self-contained)